### DISCUSSION TITLE
Wrong TPI when threading

### DISCUSSION THREAD

**Discussion by richdon1** [kachurovskiy/nanoels#189] [Q&A]
So I setup my nanoels for my lathe Z axis only, no X for now. When threading in manual mode, I have to set myTPI 23% higher then what I want to get a good thread Ex: set tpi to 16 to get a 13 tpi thread. I set my Z axis to 2400 for a 3 to 1 reduction and 800 step encoder. What am I doing wrong?

**Comment by richdon1** [comment]
I think I answered my own question. I just substracted 23% of my original Z of 2400 to 1860 and I am pretty much spot on. I will try again tomorow with different pitch threads and go from there.
Don

**Comment by kachurovskiy** [comment]
@richdon1 it probably means that your distance-per-motor-step settings need adjustment - https://github.com/kachurovskiy/nanoels/blob/main/h4/h4.ino#L14-L15 - or some of the belts or gears are slipping?

To help with fine tuning, try precision move function - first move set a dial indicator on the axis, then move a little bit in one direction to take out the backlash, zero your indicator and then do a precision move for e.g. 5mm and compare it to the indicator value.

> When off, use numpad buttons to enter a custom move distance. Pressing a move button when the screen bottom line shows `Use 1.234mm?` will move 1.234mm in the corresponding direction.

**Reply by richdon1** [reply]
i did the procedure and for a 10 mm move, my indicator shows 7.76 mm.  Now should I increase or decrease  the Z_du parameter? I ask  because everytime I want to upload a sketch, h4.ino can't find the port and it is a real pain to be able to load a sketch.

**Reply by richdon1** [reply]
I just noticed that the difference in Z is approximately the same as I had before. So, I will increase my Z to original 2400 steps and will do the procedure again and see how thar goes.
Don

**Reply by kachurovskiy** [reply]
Once you have the Z axis calibrated, it's important to also check the encoder accuracy since threading requires both to be correct.

To check encoder, show angles on screen, press Stop to zero and rotate the chuck by hand 1 turn forward. Degrees on screen should increase towards 360 and get back to 0. If not, it could be the cause of unexpected threading results.

Please also attach some photos of your build, especially how the encoder is setup and how the wires terminate. Thanks!

**Comment by richdon1** [comment]
Well, I set MOTOR_STEPS_Z = 2400. Now when I do a precision move of 10mm, I get 10.4mm so pretty close. I don't know if I should move it up or down. As for angles, after I defined the ENCODER_PPR = 1024; which is the encoder I am using, starting to move the chuck from 0 one full turn, it increments to 360 & back to 0. I did turn a few threads in alu. and they start nicrly but before the nut goes to the end, I need a spanner as if the thread is off ever so slightly. Afer that, I can thread the nut easily. Here are a few pics of my setup as well as the lines from my sketch
Don
![20240417_083210](https://github.com/kachurovskiy/nanoels/assets/99602399/ca2e1c88-e134-46e5-8039-9a9acfc4622e)
![20240417_131041](https://github.com/kachurovskiy/nanoels/assets/99602399/bc50df30-937e-47cc-8ef5-f7835dcd5c73)
![20240417_131321](https://github.com/kachurovskiy/nanoels/assets/99602399/e5b2b4c4-22e6-4247-b664-5aa071456aa1)
![20240417_131331](https://github.com/kachurovskiy/nanoels/assets/99602399/679e0b9e-e560-4d43-8e8a-38fdf885780d)

**Reply by kachurovskiy** [reply]
> I set MOTOR_STEPS_Z = 2400. Now when I do a precision move of 10mm, I get 10.4mm so pretty close

Try doing this several times in sequence and see if backlash affects the first reading. On second and consequent readings you should be getting exactly 10mm up to 0.01mm - if not, there's a problem somewhere.

Looking at your belt it seems like they aren't sufficiently tensioned which could result in inconsistent results. Consider the following suggestions:

- Switch to a direct drive (coaxial) on Z axis without the belt - your motor looks beefy enough to be sufficient 1:1
- Switch to GT3 belts - they are still very sturdy but much more flexible than HTD 5M
- Connect shielding on your wires to a fixed voltage - 5V or GND - whatever is available nearby - looks like your shields are currently disconnected

Let me know if you need help making GT3 pulleys. Cheers!

**Comment by richdon1** [comment]
My motor is a Nema 34, 8 Nm. About the shielding, the only wires I have connected to the shielding is the spindle encoder 0 volts and ground wound together on the controller groung input, Should I have more? The ac line and psu are star grounded together in the case. Do I need to connect another wire to this ground? As for precision positioning, it is all but precise. I did as you mentionned above and the results of the 1st,2nd and 3rd seqquences are in the pictures.  This is for a 5mm move. Also a very odd issue; every time  I moved the carriage when controller was in on position, it would move the specified distance then the driver would fall in fault condition. I had to unplug then replug the psu and driver. I will recheck alll wiring tomorow.
Don
PS- I would also like to mention that the only power connection to the controller is from the usb port.

![20240417_210822](https://github.com/kachurovskiy/nanoels/assets/99602399/75f290dc-96d8-499b-a24b-fe814d3f17ea)
![20240417_210909](https://github.com/kachurovskiy/nanoels/assets/99602399/1b2c42f8-ac3a-4e02-95d6-2b79f08bf3da)
![20240417_210940](https://github.com/kachurovskiy/nanoels/assets/99602399/84f1e23c-5588-4720-aa3d-d67ac6b82fe7)
